tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/brc20/BRC20TokenLibrary.sol

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import { Btcs } from "./Btcs.sol";
+import { Cats } from "./Cats.sol";
+import { Long } from "./Long.sol";
+import { Mmss } from "./Mmss.sol";
+import { Ordi } from "./Ordi.sol";
+import { Rats } from "./Rats.sol";
+import { Sats } from "./Sats.sol";
+
+library BRC0TokenLibrary {
+	function getBtcsInfo()
```

Comment:
Would mapping {string => uint256[] ranges} work, then we can get rid of the hard coded getXyzInfo, instead use the mapping dynamicly map tokenName to get the token's range?

Reply:
like this?:
```contract ERC20Mapping is ERC20{
    constructor(){
        // ada
        tokenNames['ada'] = Ada.getTokenName();
        tokenRanges['ada'] = Ada.getTokenRanges();
        tokenBscAddress['ada'] = Ada.getTokenBscAddress();
        tokenEthereumAddress['ada'] = Ada.getTokenEthereumAddress();
        // amp
        tokenNames['amp'] = Amp.getTokenName();
	tokenNames['amp'] = Amp.getTokenRanges();
	tokenNames['amp'] = Amp.getTokenBscAddress();
	tokenNames['amp'] = Amp.getTokenEthereumAddress();
} ```